providers/base/tests/test_wifi_nmcli_test.py

```diff
@@ -36,3 +51,563 @@ def test_legacy_nmcli_false(self, subprocess_mock):
+        self, get_nm_wireless_connections_mock, sp_call_mock, print_mock
+    ):
+        delete_test_ap_ssid_connection()
+
+    @patch("wifi_nmcli_test.sp.call", side_effect=Exception("Deletion failed"))
+    @patch(
+        "wifi_nmcli_test._get_nm_wireless_connections",
+        return_value={
+            "TEST_CON": {"uuid": "uuid-test", "state": "deactivated"}
+        },
+    )
+    def test_delete_test_con_exception(
+        self, get_nm_wireless_connections_mock, sp_call_mock
+    ):
+        delete_test_ap_ssid_connection()
+
+    @patch("wifi_nmcli_test._get_nm_wireless_connections", return_value={})
+    def test_no_test_con_to_delete(self, get_nm_wireless_connections_mock):
+        delete_test_ap_ssid_connection()
+
+
+class TestListAps(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_no_essid(self, check_output_mock):
+        check_output_mock.return_value = (
+            b"wlan0 \nSSID1:1:2412:60\nSSID2:6:2437:70\nSSID3:11:2462:80"
+        )
+        expected = {
+            "SSID1": {"Chan": "1", "Freq": "2412", "Signal": "60"},
+            "SSID2": {"Chan": "6", "Freq": "2437", "Signal": "70"},
+            "SSID3": {"Chan": "11", "Freq": "2462", "Signal": "80"},
+        }
+        self.assertEqual(list_aps("wlan0"), expected)
+
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_with_essid(self, check_output_mock):
+        check_output_mock.return_value = (
+            b"SSID1:1:2412:60\nSSID2:6:2437:70\nSSID3:11:2462:80"
+        )
+        expected = {
+            "SSID2": {"Chan": "6", "Freq": "2437", "Signal": "70"},
+        }
+        self.assertEqual(list_aps("wlan0", "SSID2"), expected)
+
+    @patch("wifi_nmcli_test.sp.check_output")
+    def test_list_aps_empty_output(self, check_output_mock):
+        check_output_mock.return_value = b""
+        expected = {}
+        self.assertEqual(list_aps("wlan0"), expected)
+
+
+class TestShowAps(unittest.TestCase):
+    @patch("wifi_nmcli_test.print")
+    def test_show_aps_empty(self, mock_print):
+        aps_dict = {}
+        show_aps(aps_dict)
+
+    @patch("wifi_nmcli_test.print")
+    def test_show_aps_multiple_aps(self, mock_print):
+        aps_dict = {
+            "AP1": {"Chan": "1", "Freq": "2412", "Signal": "-40"},
+            "AP2": {"Chan": "6", "Freq": "2437", "Signal": "-50"},
+        }
+        show_aps(aps_dict)
+        expected_calls = [
+            call("SSID: AP1 Chan: 1 Freq: 2412 Signal: -40"),
+            call("SSID: AP2 Chan: 6 Freq: 2437 Signal: -50"),
+        ]
+        mock_print.assert_has_calls(expected_calls, any_order=True)
+
+
+class TestWaitForConnected(unittest.TestCase):
+    @patch("wifi_nmcli_test.sp.check_output")
+    @patch("wifi_nmcli_test.print_cmd")
+    @patch("wifi_nmcli_test.time.sleep", return_value=None)
+    def test_wait_for_connected_success(
+        self, mock_sleep, mock_print_cmd, mock_check_output
+    ):
```

Comment:
If you are not using the mocks and just want to mock the calls to the function, you can just set that in the decorator:
```suggestion
    @patch("wifi_nmcli_test.print_cmd", new=MagicMock())
    @patch("wifi_nmcli_test.sp.check_output")
    @patch("wifi_nmcli_test.time.sleep", return_value=None)
    def test_wait_for_connected_success(
        self, mock_sleep, mock_check_output
    ):
```